lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex

```diff
@@ -69,39 +78,12 @@ defmodule Trento.Discovery.Payloads.Cluster.ClusterDiscoveryPayload do
     end)
   end
 
-  defp enrich_cluster_and_hana_architecture_types(attrs, "ScaleUp") do
-    attrs
-    |> Map.put("cluster_type", ClusterType.hana_scale_up())
-    |> Map.put("hana_architecture_type", HanaArchitectureType.angi())
-  end
-
-  defp enrich_cluster_and_hana_architecture_types(attrs, "ScaleOut") do
-    attrs
-    |> Map.put("cluster_type", ClusterType.hana_scale_out())
-    |> Map.put("hana_architecture_type", HanaArchitectureType.angi())
-  end
-
-  defp enrich_cluster_and_hana_architecture_types(attrs, nil) do
-    attrs
-    |> Map.put("cluster_type", parse_cluster_type(attrs))
-    |> maybe_put_hana_classic_architecture
-  end
-
-  defp parse_cluster_type(%{"crmmon" => %{"clones" => nil, "groups" => nil}}),
+  # The cluster manages a HANA system when it manages a SAPHanaTopoligy resource
+  # and other SAPHana for Scale up or SAPHanaController for Scale Out.
+  defp parse_hana_cluster_type(%{"crmmon" => %{"clones" => nil}}),
     do: ClusterType.unknown()
 
-  defp parse_cluster_type(%{"crmmon" => %{"clones" => nil, "groups" => groups}}) do
-    sap_instance_count =
-      Enum.count(groups, fn %{"resources" => resources} ->
-        Enum.any?(resources, fn %{"agent" => agent} ->
-          agent == "ocf::heartbeat:SAPInstance"
-        end)
-      end)
-
-    do_detect_cluster_type(sap_instance_count)
-  end
-
-  defp parse_cluster_type(%{"crmmon" => %{"clones" => clones}}) do
+  defp parse_hana_cluster_type(%{"crmmon" => %{"clones" => clones}}) do
```

Comment:
The most of the changes are basically due change in order of the functions in the file.
Now, instead of returning the first found type, we parse HANA and ASCS/ERS types as we did before, but based in the two values, we have the possibility to have a `hana_ascs_ers` cluster.